aiohttp/helpers.py

```diff
@@ -1038,3 +1039,19 @@ def parse_http_date(date_str: Optional[str]) -> Optional[datetime.datetime]:
             with suppress(ValueError):
                 return datetime.datetime(*timetuple[:6], tzinfo=datetime.timezone.utc)
     return None
+
+
+# Addresses can be either tuples of varying lengths (AF_INET, AF_INET6,
+# AF_NETLINK, AF_TIPC) or strings (AF_UNIX).
+def connect_socket_with_interface(
+    network_interface: str,
+    family: int = socket.AF_INET,
+    local_addr: Optional[Union[Tuple[Any, ...], str]] = None,
+) -> socket.socket:
+    """Prepares a socket to asyncio"""
```

Comment:
The docstrings must follow PEP 257. This means having a full stop after the title-sentence. Also, the title must be in an imperative mood ("Do smth", not "Does smth") — similar to Git commit best practices.
Additionally, the docstring should be accurate including what's the effect of calling the function — not an abstract action on an abstract thing.